core/services/fluxmonitor/flux_monitor.go

```diff
@@ -455,8 +475,75 @@ func (p *PollingDeviationChecker) consume(ctx context.Context, roundSubscription
 	}
 }
 
-func (p *PollingDeviationChecker) pollIfEligible(client eth.Client, threshold float64) bool {
-	open, err := p.isEligibleToPoll(client)
+func (p *PollingDeviationChecker) respondToLog(log interface{}) (roundStarted bool, roundEnded bool, _ error) {
+	switch log := log.(type) {
+	case contracts.LogNewRound:
+		return true, false, p.respondToNewRoundLog(log)
+	case contracts.LogRoundDetailsUpdated:
+		return false, false, p.respondToRoundDetailsUpdatedLog(log)
+	case contracts.LogAnswerUpdated:
+		return false, true, p.respondToAnswerUpdatedLog(log)
+	default:
+		panic("got unknown log")
```

Comment:
Happy that this is changed in a subsequent pr :+1:

Reply:
Went ahead and pushed the change here as well.